src/common/lib/client/rest.ts

```diff
@@ -228,6 +230,155 @@ class Rest {
     }
   }
 
+  batchPublish(
+    specs: API.Types.BatchPublishSpec,
+    callback: API.Types.StandardCallback<
+      API.Types.BatchResult<API.Types.BatchPublishSuccessResult | API.Types.BatchPublishFailureResult>
+    >
+  ): void;
+  batchPublish(
+    specs: API.Types.BatchPublishSpec[],
+    callback: API.Types.StandardCallback<
+      API.Types.BatchResult<API.Types.BatchPublishSuccessResult | API.Types.BatchPublishFailureResult>[]
+    >
+  ): void;
+  batchPublish(
+    specs: API.Types.BatchPublishSpec
+  ): Promise<API.Types.BatchResult<API.Types.BatchPublishSuccessResult | API.Types.BatchPublishFailureResult>>;
+  batchPublish(
+    specs: API.Types.BatchPublishSpec[]
+  ): Promise<API.Types.BatchResult<API.Types.BatchPublishSuccessResult | API.Types.BatchPublishFailureResult>[]>;
+  batchPublish(
```

Comment:
could simplify this signature a tiny bit using generics
```typescript
  batchPublish<T extends BatchPublishSpec | BatchPublishSpec[]>(specs: T): void 
  | Promise<T extends BatchPublishSpec ? BatchResult<BatchPublishSuccess | BatchPublishFailure> : BatchResult<BatchPublishSuccess | BatchPublishFailure>[]>
```

Reply:
Nice, thanks! Have updated.